docs/dev-guide/tscript/messages.mdx

```diff
@@ -0,0 +1,440 @@
+
+A Temporal Client can use [Signal with Start](#signal-with-start) to kick off a Workflow Execution.
+
+See also: [Common problems with Signals and their Workarounds](https://github.com/temporalio/edu-interacting-with-workflows-typescript-content/blob/main/signaling-your-workflows/common-issues-with-signals.md)
+
+### Define a Signal {#define-signal}
+
+A Signal has a name and can have arguments.
+
+- The name, also called a Signal type, is a case-sensitive string.
+- The name is used to identify the Signal when the Client communicates with the Temporal Service.
+- The arguments must be [serializable](/dataconversion).
+- Each Workflow Definition can support multiple Signals.
+
+Use the [`defineSignal`](https://typescript.temporal.io/api/namespaces/workflow/#definesignal) function to create your Signals:
+
+```ts
+import { defineSignal } from '@temporalio/workflow';
+
+interface JoinInput {
+  userId: string;
+  groupId: string;
+}
+
+export const joinSignal = defineSignal<[JoinInput]>('join');
+```
+
+### Handle a Signal {#handle-signal}
+
+Workflows listen for Signals by the Signal's name.
+Signal handlers are functions defined in the Workflow that listen for incoming Signals of a given type.
+These handlers define how a Workflow should react when it receives a specific type of Signal.
+To implement a Signal handler, you will need to use the [`setHandler`](https://typescript.temporal.io/api/namespaces/workflow/#sethandler) function provided by the TypeScript SDK API.
+
+```ts
+import { setHandler } from '@temporalio/workflow';
+
+export async function yourWorkflow() {
+  const groups = new Map<string, Set<string>>();
+
+  setHandler(joinSignal, ({ userId, groupId }: JoinInput) => {
+    const group = groups.get(groupId);
+    if (group) {
+      group.add(userId);
+    } else {
+      groups.set(groupId, new Set([userId]));
+    }
+  });
+}
+```
+
+The Workflow dynamically reacts to external events without interrupting its execution flow.
+There can be only one handler for any Signal type; calling `setHandler` multiple times for the same Signal type overwrites the existing handler.
+For example, if you first set Handler A for a given Signal, and then later in the Workflow Definition set Handler B for the same Signal, then only Handler B will be invoked upon receiving this Signal.
+
+### Send a Signal from a Temporal Client {#send-signal-from-client}
+
+**How to send a Signal from a Temporal Client**
+
+When a Signal is sent successfully from the Temporal Client, the [WorkflowExecutionSignaled](/references/events#workflowexecutionsignaled) Event appears in the Event History of the Workflow that receives the Signal.
+
+<!--come back and review this and reword-->
+The [`WorkflowHandle.signal`](https://typescript.temporal.io/api/interfaces/client.WorkflowHandle#signal) method synchronously awaits the signal sending?
```

Comment:
Maybe move this line (line 119) to under the description of the handle (line 132) and write:

The [WorkflowHandle.signal](https://typescript.temporal.io/api/interfaces/client.WorkflowHandle#signal) method is used to send the Signal to the handle of the Workflow.

Reply:
Updated